File: src/main/java/com/grizzly/baseViews/BaseLinearLayout.java
package com.grizzly.baseViews;

import android.annotation.TargetApi;
import android.app.Activity;
import android.content.Context;
import android.os.Build;
import android.support.v4.view.AsyncLayoutInflater;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

/**
 * Created by FcoPardo on 3/6/16.
 *
 * Base class for creating custom views using a LinearLayout.
 */
public abstract class BaseLinearLayout extends LinearLayout {

    protected int layout = 0;

    public BaseLinearLayout(Context context) {
        super(context);
        inflateBaseLayout();
    }

    public BaseLinearLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        inflateBaseLayout();
    }

    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
    public BaseLinearLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        inflateBaseLayout();
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public BaseLinearLayout(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        inflateBaseLayout();
    }

    protected void inflateBaseLayout(){
        setContainer();
        if(layout>0){
            //BaseView.inflateLayout(layout, getContext(), this);
            BaseView.inflateLayout(layout, getContext(), this, new AsyncLayoutInflater.OnInflateFinishedListener() {
                public void onInflateFinished(View view, int resid, ViewGroup parent) {
                    inflateComponents();
                }
            });
        }
        //inflateComponents();
    }

    protected abstract void inflateComponents();

    protected abstract void setContainer();

    protected Activity getActivity(){
        return BaseView.getActivity(this);
    }
}

File: src/main/kotlin/com/grizzly/baseViews/BaseLinearLayout.kt
package com.grizzly.baseViews

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout

/**
 * Created by FcoPardo on 4/29/18.
 */
open abstract class BaseLinearLayout : LinearLayout, BaseView.OnVariableView {

    override var layout = 0

    constructor(context : Context) : super(context){
        inflateBaseLayout()
    }
    constructor(context : Context, attrs : AttributeSet) : super(context, attrs) {
        inflateBaseLayout()
    }
    constructor(context : Context, attrs : AttributeSet, styleAttr : Int) : super(context, attrs, styleAttr){
        inflateBaseLayout()
    }
    constructor(context : Context, attrs : AttributeSet, styleAttr : Int, defStyleAttr : Int) : super(context, attrs, styleAttr, defStyleAttr){
        inflateBaseLayout()
    }

}